cmd/sonobuoy/app/images.go

```diff
@@ -101,3 +157,195 @@ func getImages(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 }
+
+func pullImages(cmd *cobra.Command, args []string) {
+	switch imagesflags.plugin {
+	case "e2e":
+
+		cfg, err := imagesflags.kubeconfig.Get()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get REST client"))
+			os.Exit(1)
+		}
+
+		sbc, err := getSonobuoyClient(cfg)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "could not create sonobuoy client"))
+			os.Exit(1)
+		}
+
+		version, err := sbc.Version()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get Sonobuoy client"))
+			os.Exit(1)
+		}
+
+		upstreamImages, err := image.GetImages("", version)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init upstream registry list"))
+			os.Exit(1)
+		}
+
+		for _, v := range upstreamImages {
+			err := docker.PullIfNotPresent(v.GetE2EImage(), numDockerRetries)
+			if err != nil {
+				errlog.LogError(errors.Wrapf(err, "couldn't pull image: %v", v.GetE2EImage()))
+			}
+		}
+	default:
+		errlog.LogError(errors.Errorf("Unsupported plugin: %v", imagesflags.plugin))
+		os.Exit(1)
+	}
+}
+
+func downloadImages(cmd *cobra.Command, args []string) {
+	switch imagesflags.plugin {
+	case "e2e":
+
+		cfg, err := imagesflags.kubeconfig.Get()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get REST client"))
+			os.Exit(1)
+		}
+
+		sbc, err := getSonobuoyClient(cfg)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "could not create sonobuoy client"))
+			os.Exit(1)
+		}
+
+		version, err := sbc.Version()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get Sonobuoy client"))
+			os.Exit(1)
+		}
+
+		upstreamImages, err := image.GetImages("", version)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init upstream registry list"))
+			os.Exit(1)
+		}
+
+		images := []string{}
+		for _, v := range upstreamImages {
+			images = append(images, v.GetE2EImage())
+		}
+		err = docker.Save(images, image.GetTarFileName(version))
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't save images to tar"))
+			os.Exit(1)
+		}
+
+	default:
+		errlog.LogError(errors.Errorf("Unsupported plugin: %v", imagesflags.plugin))
+		os.Exit(1)
+	}
+}
+
+func pushImages(cmd *cobra.Command, args []string) {
+
+	switch imagesflags.plugin {
+	case "e2e":
+
+		if len(imagesflags.e2eRegistryConfig) > 0 {
+			// Check if the e2e file exists
+			if _, err := os.Stat(imagesflags.e2eRegistryConfig); os.IsNotExist(err) {
+				errlog.LogError(errors.Errorf("file does not exist or cannot be opened: %v", imagesflags.e2eRegistryConfig))
```

Comment:
same nit as the other line like this; by checking the `isNotExist` we aren't really checking the `cannot be opened` case. Maybe just remove that wording from the error in each case?

Reply:
The os.Stat check if we can open I think. We're creating the file and overwriting each time as well.

Reply:
Sorry, I'm not sure I understand your response.

I agree it should check if the file exists or we cant open it. But if you check `os.IsNotExist` then you're only checking whether or not the file exists. Other sorts of access issues wont hit this (and the other similar) blocks. Right?